pkg/new_storage/s3.go

```diff
@@ -59,15 +61,29 @@ func (s *S3) Connect() error {
 
 	awsDefaults := defaults.Get()
 	defaultCredProviders := defaults.CredProviders(awsDefaults.Config, awsDefaults.Handlers)
+	customCredProviders := defaultCredProviders
 
-	// Define custom static cred provider
-	staticCreds := &credentials.StaticProvider{Value: credentials.Value{
-		AccessKeyID:     s.Config.AccessKey,
-		SecretAccessKey: s.Config.SecretKey,
-	}}
+	if s.Config.AccessKey != "" && s.Config.SecretKey != "" {
+		// Define custom static cred provider
+		staticCreds := &credentials.StaticProvider{Value: credentials.Value{
+			AccessKeyID:     s.Config.AccessKey,
+			SecretAccessKey: s.Config.SecretKey,
+		}}
+		// Append static creds to the defaults
+		customCredProviders = append([]credentials.Provider{staticCreds}, customCredProviders...)
+	}
+
+	awsRoleARN := os.Getenv("AWS_ROLE_ARN")
```

Comment:
Is these credentials stored only on enviroment variables or can be passed as config parameter?

Reply:
> Is these credentials stored only on enviroment variables or can be passed as config parameter?

They are automatically added to the environment by Kubernetes.